fabric-entity-events-v1/src/client/java/net/fabricmc/fabric/api/entity/event/client/ClientPlayerEvents.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.entity.event.client;
+
+import net.minecraft.client.network.ClientPlayerEntity;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class ClientPlayerEvents {
+	/*
+	 * Flag for using default slowdown during using an item.
+	 */
+	public static final float USE_DEFAULT_SLOWDOWN_SPEED = -1.0F;
+
+	/*
+	 * Default percentage of speed slowdown when using an item in Minecraft.
+	 */
+	private static final float DEFAULT_SLOWDOWN_SPEED = 0.2F;
+
+	/**
+	 * An event that is called when a player is moving during using an item.
+	 */
+	public static final Event<AdjustUsingItemSpeed> ADJUST_USING_ITEM_SPEED = EventFactory.createArrayBacked(AdjustUsingItemSpeed.class, callbacks -> player -> {
+		float maxSpeed = USE_DEFAULT_SLOWDOWN_SPEED;
+
+		for (AdjustUsingItemSpeed callback : callbacks) {
+			float currentSpeed = callback.adjustUsingItemSpeed(player);
+
+			if (currentSpeed >= 0.0F && currentSpeed <= 1.0F) {
```

Comment:
why is the speed capped at 1.0?

Reply:
On a similar note, why doesn't it accept speeds below zero? Is it just because of the `-1.0F` fallback or do negative values crash the game? If it's the former, consider allowing negative values after changing the fallback to `@Nullable Float`. If it's the latter, consider adding a comment to document why negative values would break stuff.

Reply:
Since this event is used to adjust the player's movement speed while using items, a value of 0 means the player's speed is already 0. A value of 1 means the player will maintain their original movement speed without any slowdown.
When the value is negative, it implies that the player will actually move backward when trying to move forward, which is counterintuitive. Therefore, negative values are not needed. 
When the value is greater than 1, it can be understood as the player moving faster while using an item. I'm currently questioning whether this is even necessary.

Reply:
Personally, I find it perfectly intuitive that negative speed would move you backwards. Do we really ever know that this functionality will not be necessary for any current or future mod? It just seems like an artificial limit to me.

If it requires extra effort to make these values work, that would be a different story. But given that it seems to just work without the limit, I think we might as well remove the limit and give modders the flexibility to use crazy values if they want to.

Reply:
Thank you! I understand, and I will remove this limit.